src/packageurl/contrib/purl2url.py

```diff
@@ -28,6 +28,19 @@
 from packageurl.contrib.route import NoRouteAvailable
 from packageurl.contrib.route import Router
 
+
+def get_repo_download_url_by_package_type(type, namespace, name, version):
+    """
+    Return a download URL for package repo by package type.
+    """
+    download_url_by_type = {
+        "github": f"https://github.com/{namespace}/{name}/archive/refs/tags/{version}.zip",
+        "bitbucket": f"https://bitbucket.org/{namespace}/{name}/get/{version}.zip",
+        "gitlab": f"https://gitlab.com/{namespace}/{name}/-/archive/{version}/{name}-{version}.zip",
+    }
+    return download_url_by_type.get(type)
```

Comment:
Add to the docstring that you can return None too.

Also you could have an argument to fetch "zip" or "tar.gz" archives. I think we will get better support across for links and similar as well as smaller files with tar.gz. All three support this:
may something like this:
```Python
def get_repo_download_url_by_package_type(type, namespace, name, version, archive_extension="tar.gz"):
    """
    Return the download URL for a hosted git repository given a package type
    or None.
    """
    assert archive_extension in (".zip", ".tar.gz",) 
    download_url_by_type = {
        "github": f"https://github.com/{namespace}/{name}/archive/refs/tags/{version}.{archive_extension}",
        "gitlab": f"https://gitlab.com/{namespace}/{name}/-/archive/{version}/{name}-{version}.{archive_extension}",
        "bitbucket": f"https://bitbucket.org/{namespace}/{name}/get/{version}.{archive_extension}",
    }
    return download_url_by_type.get(type)
```

A side note for the future: we may want to avoid computing all URLs for all types ahead of time

Reply:
done!